nym-vpn-core/crates/nym-vpn-account-controller/src/commands/zknym_handler/request.rs

```diff
@@ -98,24 +167,42 @@ impl RequestZkNymTask {
         let poll_result = self.poll_zk_nym(&id).await?;
         let ticketbook_type = poll_result.ticketbook_type.clone();
 
-        // The result might contain attached keys and signatures. If so, import them.
-        self.import_attached_keys_and_signatures(&poll_result, pending_request.expiration_date)
-            .await?;
-
-        // Import the zk-nym ticketbook itself. This will unblind and aggregate the zk-nym shares
-        self.import_zk_nym(poll_result, pending_request).await?;
+        let success = match poll_result.status {
+            NymVpnZkNymStatus::Pending => {
+                unreachable!("poll_zk_nym would not return an Ok with Pending status")
+            }
+            NymVpnZkNymStatus::Revoking | NymVpnZkNymStatus::Revoked => {
+                return Err(RequestZkNymError::ZkNymRevoked);
+            }
+            // unfortunately no error message is included, so we just replicate old behaviour
+            NymVpnZkNymStatus::Error => {
+                return Err(RequestZkNymError::MissingBlindedShares);
```

Comment:
While that whole "checking the zk-nym status" part was very much missing and leading to weird errors before, we should definitely not return that error there. 
It should be a new variant, akin to the `Revoked` one above.